lib/url.js

```diff
@@ -602,13 +602,18 @@ Url.prototype.format = function() {
 
   // only the slashedProtocols get the //.  Not mailto:, xmpp:, etc.
   // unless they had them to begin with.
-  if (this.slashes ||
-      (!protocol || slashedProtocol[protocol]) && host !== false) {
-    host = '//' + (host || '');
-    if (pathname && pathname.charCodeAt(0) !== 47/*/*/)
-      pathname = '/' + pathname;
-  } else if (!host) {
-    host = '';
+  if (this.slashes || slashedProtocol[protocol]) {
+    if (this.slashes || host) {
+      if (pathname && pathname.charCodeAt(0) !== 47/*/*/)
+        pathname = '/' + pathname;
+      host = '//' + host;
+    } else if (protocol.length >= 4 &&
+               protocol.charCodeAt(0) === 102/*f*/ &&
```

Comment:
@indutny I used decimal because:

* It's decimal most of the existing `charCodeAt()` calls, including ~10 lines later in lines 621 and 622, as well as the `switch` statement ~30 lines earlier starting on line 581.
* The sample code (provided by @mscdex) used decimal and I did a lazy copy/paste.

I don't feel strongly about it, though, and I'm happy to change it to hex format if that's the difference between a quick `LGTM` and a not-sure-about-this.